File: gencode_node/todo_1_gpt-5.4.js
import { createServer } from 'node:http'
import { randomUUID } from 'node:crypto'

const todos = []

const send = (res, status, data) => {
  res.writeHead(status, {
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Methods': 'GET,POST,PATCH,DELETE,OPTIONS',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Content-Type': 'application/json',
  })
  res.end(JSON.stringify(data))
}

const readBody = (req) =>
  new Promise((resolve, reject) => {
    let body = ''
    req.on('data', (chunk) => (body += chunk))
    req.on('end', () => resolve(body ? JSON.parse(body) : {}))
    req.on('error', reject)
  })

createServer(async (req, res) => {
  const url = new URL(req.url || '/', 'http://localhost')
  const id = url.pathname.split('/')[2]
  const todo = todos.find((item) => item.id === id)

  try {
    if (req.method === 'OPTIONS') return send(res, 204, {})
    if (url.pathname === '/todos' && req.method === 'GET') return send(res, 200, todos)
    if (url.pathname === '/todos' && req.method === 'POST') {
      const { text = '' } = await readBody(req)
      if (!text.trim()) return send(res, 400, { error: 'text is required' })
      const newTodo = { id: randomUUID(), text: text.trim(), completed: false, createdAt: Date.now() }
      todos.push(newTodo)
      return send(res, 201, newTodo)
    }
    if (url.pathname.startsWith('/todos/') && req.method === 'GET') {
      return todo ? send(res, 200, todo) : send(res, 404, { error: 'todo not found' })
    }
    if (url.pathname.startsWith('/todos/') && req.method === 'PATCH') {
      if (!todo) return send(res, 404, { error: 'todo not found' })
      const { text, completed } = await readBody(req)
      if (typeof text === 'string') todo.text = text.trim() || todo.text
      if (typeof completed === 'boolean') todo.completed = completed
      return send(res, 200, todo)
    }
    if (url.pathname.startsWith('/todos/') && req.method === 'DELETE') {
      const index = todos.findIndex((item) => item.id === id)
      if (index === -1) return send(res, 404, { error: 'todo not found' })
      const [deleted] = todos.splice(index, 1)
      return send(res, 200, deleted)
    }
    return send(res, 404, { error: 'route not found' })
  } catch {
    return send(res, 400, { error: 'invalid request body' })
  }
}).listen(3001, () => console.log('Todo API running on http://localhost:3001'))

File: gencode_python/todo_1_gpt-5.4.py
import json
import time
import uuid
from http.server import BaseHTTPRequestHandler, HTTPServer

TODOS = []


def send(handler, status, payload=None):
    body = b"" if payload is None else json.dumps(payload).encode()
    handler.send_response(status)
    handler.send_header("Content-Type", "application/json")
    handler.send_header("Access-Control-Allow-Origin", "*")
    handler.send_header("Access-Control-Allow-Headers", "Content-Type")
    handler.send_header("Access-Control-Allow-Methods", "GET, POST, PATCH, DELETE, OPTIONS")
    handler.end_headers()
    if body:
        handler.wfile.write(body)


def read_json(handler):
    size = int(handler.headers.get("Content-Length", "0"))
    raw = handler.rfile.read(size) if size else b"{}"
    return json.loads(raw)


class TodoHandler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        send(self, 204)

    def do_GET(self):
        if self.path == "/todos":
            return send(self, 200, TODOS)
        send(self, 404, {"error": "Not found"})

    def do_POST(self):
        if self.path != "/todos":
            return send(self, 404, {"error": "Not found"})
        data = read_json(self)
        text = str(data.get("text", "")).strip()
        if not text:
            return send(self, 400, {"error": "text is required"})
        todo = {
            "id": uuid.uuid4().hex,
            "text": text,
            "completed": False,
            "createdAt": int(time.time() * 1000),
        }
        TODOS.insert(0, todo)
        send(self, 201, todo)

    def do_PATCH(self):
        todo = self.find_todo()
        if not todo:
            return send(self, 404, {"error": "Todo not found"})
        data = read_json(self)
        if "text" in data:
            todo["text"] = str(data["text"]).strip() or todo["text"]
        if "completed" in data:
            todo["completed"] = bool(data["completed"])
        send(self, 200, todo)

    def do_DELETE(self):
        todo = self.find_todo()
        if not todo:
            return send(self, 404, {"error": "Todo not found"})
        TODOS.remove(todo)
        send(self, 200, {"deleted": todo["id"]})

    def find_todo(self):
        parts = self.path.strip("/").split("/")
        if len(parts) != 2 or parts[0] != "todos":
            return None
        return next((item for item in TODOS if item["id"] == parts[1]), None)

    def log_message(self, *_):
        return


if __name__ == "__main__":
    print("Serving todos at http://127.0.0.1:8000")
    HTTPServer(("127.0.0.1", 8000), TodoHandler).serve_forever()